src/safeds/data/tabular/containers/_table.py

```diff
@@ -969,12 +999,14 @@ def slice_rows(
         if end is None:
             end = self.number_of_rows
 
-        if start < 0 or end < 0 or start >= self.number_of_rows or end > self.number_of_rows or end < start:
-            raise ValueError("The given index is out of bounds")
+        if end < start:
+            raise IndexOutOfBoundsError(slice(start, end))
+        if start < 0 or end < 0 or start > self.number_of_rows or end > self.number_of_rows:
+            raise IndexOutOfBoundsError(start if start < 0 or start > self.number_of_rows else end)
 
         new_df = self._data.iloc[start:end:step]
         new_df.columns = self._schema.column_names
-        return Table._from_pandas_dataframe(new_df)
+        return Table._from_pandas_dataframe(new_df, self._schema)
```

Comment:
Why would you keep the old schema on a new table?

Reply:
```suggestion
        return Table._from_pandas_dataframe(new_df)
```